src/api/app/models/staging_workflow.rb

```diff
@@ -1,4 +1,14 @@
 class StagingWorkflow < ApplicationRecord
   belongs_to :project, inverse_of: :staging
   has_many :staging_projects, class_name: 'Project', inverse_of: :staging_workflow, dependent: :nullify
+
+  has_many :staged_requests, class_name: 'BsRequest', through: :staging_projects
+
+  def unassigned_requests
+    project.target_of_bs_requests - staged_requests - ignored_requests
```

Comment:
[nitpick] You can also define an association on the StagingWorkflow
```
has_many :target_of_bs_requests, through: project
```

Reply:
Nice! :heart:  `target_of_bs_requests` better than what we had before.

But it comes to my mind that we also talked about getting only the requests that are in a "valid" state instead of all of them. As "valid" I mean requests that can be assigned to staging projects. As far as I know, those with state `new` and `review`... any other?

The scope `in_states` defined in BsRequest model could help:
`scope :in_states, ->(states) { where(state: states) } `

What do you think?

Reply:
@ChrisBr:

> [nitpick] You can also define an association on the StagingWorkflow [...]

Defined. And used. Please have a look.

@saraycp:

> [...] As "valid" I mean requests that can be assigned to staging projects. As far as I know, those with state `new` and `review`... any other?

Not any that I am aware of. Added the `in_states` scope with valid states in the query. Please have a look.